crates/wysiwyg/src/composer_model/format.rs

```diff
@@ -51,7 +53,189 @@ where
 
     pub fn inline_code(&mut self) -> ComposerUpdate<S> {
         self.push_state_to_history();
-        self.format_or_unformat(InlineFormatType::InlineCode)
+        let format_type = InlineFormatType::InlineCode;
+        if self.action_is_reversed(format_type.action()) {
+            self.unformat(format_type)
+        } else {
+            self.add_inline_code()
+        }
+    }
+
+    fn add_inline_code(&mut self) -> ComposerUpdate<S> {
+        let (s, e) = self.safe_selection();
+        let format = InlineFormatType::InlineCode;
+
+        if s == e {
+            self.toggle_zero_length_format(&format);
+            ComposerUpdate::update_menu_state(
+                self.compute_menu_state(MenuStateComputeType::KeepIfUnchanged),
+            )
+        } else {
+            let range = self.state.dom.find_range(s, e);
+            let leaves = range.leaves();
+            // We'll iterate through the leaves finding their closest structural node ancestor and
+            // grouping these leaves based on the handles of these ancestors.
```

Comment:
This should probably be a separate method

Reply:
Yup. Will help @Velin92 with the link implementation too.